components/feed/localindex.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+
+import getSpotifyData from "@/lib/getSpotifyData";
+
+import FeedItem from "./FeedItem";
+import FeedTextBox from "./FeedTextBox";
+
+import type { FirestoreFeedItem } from "@/lib/firestore/types";
+import type { User } from "next-auth";
+
+const isUserFollowing = async (username: string) =>
+  getSpotifyData<SpotifyApi.UserFollowsUsersOrArtistsResponse>(
+    `https://api.spotify.com/v1/me/following/contains?type=user&ids=${username}`
+  ).then((response) => response[0]);
+
+export type FeedCommentType = {
+  id: string;
+} & FirestoreFeedItem;
+
+export type FirestoreFeedItemWithComments = FirestoreFeedItem & {
+  comments: FeedCommentType[];
+};
+
+export type FeedItemType = {
+  id: string;
+  comments: FeedCommentType[];
+} & FirestoreFeedItemWithComments;
+
+type FilteredFeedItemType = FeedItemType;
```

Comment:
all of this is duplicated from the normal feed and should be extracted into a third file for reuse